### ISSUE TITLE
[RRFC] `npm run build:*`

### ISSUE THREAD

**Issue by askirmas** [npm/rfcs#168]
# What / Why
package.json:
```json
{
    "build:project": "",
    "build:static": "",
    "build:markdown": "",
    "build:docs": "",
    "build:export": "",
    "build": "npm run build:project && npm run build:static && npm run build:markdown && npm run build:docs && npm run something"
}
```
Idea is to avoid manual typing. In the example `build:export` not a part of `build` and `something` 
script not exists. Or deletes everything - also possible. Any human mistake on routine. 

In addition, such feature will require good naming and placement in script names for easier lifecycle handling. I.e. `build:clear` should be first, `build:export` last and `build:clear` should delete only build stuff, not launch server and make a commit. 

## When
On `npm run`

## Where
`npm run`, *bash*
### Current Behavior

`npm ERR! missing script: build:*`

### Expected Behavior
Run consistently `&&` all scripts from *package.json* without any sort that starts with given substring

Approximately:
```typescript
const pattern: string = runScript
pattern.length--
const scripts2run = Object.keys(packageJson.scripts)
.filter(scr => scr.startsWith(pattern))
.forEach(scr => child_process(`npm run ${scr}`))
```
## Who

* n/a

## References
https://github.com/npm/cli/issues/1440

**Comment by balupton** [comment]
Note that the implementation here should use the keys in their order, as well as not run items in the next tier: e.g. `npm run build:*` should not run `npm run build:compile:next`, as the latter should be run by `npm run build:compile:*`. Perhaps having `npm run build:**` for nested, same as globstar

**Comment by ljharb** [comment]
There’s no such thing as a “tier” - colons are just a non-universal convention. I use dashes in my own projects.

A star should mean what a star means; the only exception, ofc, is if the script containing the star is matched by the star, it would not be run again.

**Comment by balupton** [comment]
> There’s no such thing as a “tier” - colons are just a non-universal convention. I use dashes in my own projects.
> 
> A star should mean what a star means; the only exception, ofc, is if the script containing the star is matched by the star, it would not be run again.

depends on which non-universal convention one is using

**Comment by ljharb** [comment]
Yes, but you’d avoid that problem by using pre and post scripts anyways so that the glob ended up doing what you wanted.

There is no universal convention, and it would be inappropriate for npm to impose one.

**Comment by balupton** [comment]
> Yes, but you’d avoid that problem by using pre and post scripts anyways so that the glob ended up doing what you wanted.

more complexity, less control, less explicit order

> There is no universal convention, and it would be inappropriate for npm to impose one.

then why are you here?

**Comment by ljharb** [comment]
You’re getting unnecessarily defensive.

Pre and post scripts gives you an explicit order already. I’d suggest trying them out if you’re having issues with maintaining npm run-scripts.

I am here because i like the globbing suggestion, and i don’t like anything that would enshrine colons.

**Comment by travi** [comment]
i've used [npm-run-all](https://www.npmjs.com/package/npm-run-all) for a while, which has behavior very similar to what is being described here. seems like it should at least be considered as prior art.

**Comment by askirmas** [comment]
>  `npm run build:*` should not run `npm run build:compile:next`

`*` means `/[\w]*/`, `**` means `/.*/`. Enough for end of the universe

**Comment by askirmas** [comment]
And `npm run /<RegEx>/` to finish everything
